src/clj/puppetlabs/services/config/puppet_server_config_service.clj

```diff
@@ -26,7 +26,9 @@
           "Initializing with the following settings from core Puppet:\n%s"
           (ks/pprint-to-string puppet-config))
         (core/init-webserver! override-webserver-settings!
-                              (get-in tk-config [:webserver])
+                              (get-in tk-config
+                                      [:webserver :puppet-server]
+                                      (get-in tk-config [:webserver]))
                               puppet-config)
```

Comment:
I'm good with this as it stands for now.  Separate tk-j9 improvement for later sounds like a good thing, too, although I'm still not sure what an improvement to it which would solve the problem we have here would look like.  Just wish we could get rid of all of this override stuff....